sdk/eventhub/azure-eventhub/azure/eventhub/_consumer_client.py

```diff
@@ -392,7 +469,30 @@ def _receive(
                 except KeyError:
                     pass
 
-    def receive(self, on_event: Callable[["PartitionContext", Optional["EventData"]], None], **kwargs: Any) -> None:
+    def receive( # pylint:disable=unused-argument
+            self,
+            on_event: Callable[["PartitionContext", Optional["EventData"]], None],
+            *,
+            max_wait_time: Optional[float] = None,
+            partition_id: Optional[str] = None,
+            owner_level: Optional[int] = None,
+            prefetch: int = 300,
+            track_last_enqueued_event_properties: bool = False,
+            starting_position: Optional[
+                Union[str, int, datetime.datetime, Dict[str, Any]]
+            ] = None,
```

Comment:
I left it as is because while the docstring doesn't say it, our typing does and if someone was passing in None in their code it would end up breaking.